### ISSUE TITLE
Transform methods with union type as arguments to overloaded methods

### ISSUE THREAD

**Issue by MangelMaxime** [glutinum-org/cli#29]
TypeScript documentation: 

```ts
export class Logger {
    log(value : string | number): void
}
```

translates into

```fs
module rec Glutinum

open Fable.Core
open System

[<Erase>]
type Exports =
    interface end

[<AllowNullLiteral>]
type Logger =
    abstract member log: value: U2<string, float> -> unit
```

We should also tries to convert union type coming from type reference:

```ts
type Value = string | number

export class Logger {
    log(value : Value): void
}
```

translates into

```fs
module rec Glutinum

open Fable.Core
open System

[<Erase>]
type Exports =
    interface end

type Value =
    U2<string, float>

[<AllowNullLiteral>]
type Logger =
    abstract member log: value: string -> unit
    abstract member log: value: float -> unit
//    abstract member log: value: Value -> unit // Should we keep this version ?
```

~The `Value` type should probably be kept if the user needs to write helpers or API depending on that type.~

For the first implementation, we are going to try removing the type reference declaration if only used inside of method overloading.

Example of npm package using this features:

[Day.js](https://github.com/iamkun/dayjs/blob/bdcc336613c9fa466b385574e88d0b43629475bc/types/index.d.ts#L415)

**Comment by goswinr** [comment]
My intuition is to not have the erased union initially.
It could always be added later, but removing it later would be a braking change.

**Comment by MangelMaxime** [comment]
I don't think we will always be able to remove that type definition. For example, if the type reference i used as the type of a property we will not be able to remove it.

But I updated the original issue to try removing the type reference and we will see where it cause problems.

```ts
type PostalCode = string | number

export class City {
	postalCode : PostalCode
}

export class Person {
	postalCode : PostalCode
}
```

**Comment by goswinr** [comment]
Yes, if a type is used as a return value in the same library or as property then it can not be split into overloads only. But I had the impression such types are most often used just as arguments. 
It is then basically the same as using overloads.
